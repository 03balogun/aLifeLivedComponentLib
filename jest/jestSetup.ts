/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
