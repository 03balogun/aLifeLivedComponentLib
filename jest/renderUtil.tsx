import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
