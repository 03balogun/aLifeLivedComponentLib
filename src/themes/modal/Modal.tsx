import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
// Components
import { Box } from '../Box';

type Props = {
  children: React.ReactNode;
  onPress: () => void;
};

export const Modal = ({ onPress, children }: Props) => {
  return (
    <TouchableWithoutFeedback onPress={() => onPress()}>
      <Box flex={1} width="100%" justifyContent="center">
        <Box backgroundColor="primaryHeavy" margin="s" borderRadius="medium">
          {children}
        </Box>
      </Box>
    </TouchableWithoutFeedback>
  );
};
