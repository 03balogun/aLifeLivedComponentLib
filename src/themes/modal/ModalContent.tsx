import React from 'react';
import { Box } from '../Box';
import { Text } from '../text/Text';

type ModalHeaderProps = {
  header: string;
  description?: string;
};

export const ModalContent: React.FC<ModalHeaderProps> = ({
  header,
  description,
}) => {
  return (
    <Box m="m" paddingBottom="s" alignContent="center" alignItems="center">
      <Text variant="h2" fontWeight="bold">
        {header}
      </Text>
      {description && (
        <Text variant="body" marginTop="m">
          {description}
        </Text>
      )}
    </Box>
  );
};
