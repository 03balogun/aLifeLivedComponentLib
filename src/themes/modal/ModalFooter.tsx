import React from 'react';
import { Box } from '../Box';

type ModalHeaderProps = {
  children: React.ReactNode;
};

export const ModalFooter: React.FC<ModalHeaderProps> = props => {
  return (
    <Box margin="m" flexDirection="row" alignSelf="center" flexWrap="wrap">
      {props.children}
    </Box>
  );
};
