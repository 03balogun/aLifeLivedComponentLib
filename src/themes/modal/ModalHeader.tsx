import React from 'react';
import { Box } from '../Box';
import { SystemPressable } from '../systemPressable/SystemPressable';
import { Text } from '../text/Text';

type ModalHeaderProps = {
  onHeaderPress: () => void;
};

export const ModalHeader: React.FC<ModalHeaderProps> = ({ onHeaderPress }) => {
  return (
    <Box m="s" alignContent="flex-start" alignItems="flex-start">
      <SystemPressable onPress={() => onHeaderPress()}>
        <Text variant="h3" fontWeight="bold">
          X
        </Text>
      </SystemPressable>
    </Box>
  );
};
