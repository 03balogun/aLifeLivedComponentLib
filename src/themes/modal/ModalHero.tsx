import React from 'react';
import { Box } from '../Box';
import { Text } from '../text/Text';

type ModalHeroProps = {
  header: string;
};

export const ModalHero: React.FC<ModalHeroProps> = props => {
  return (
    <Box m="s">
      <Text variant="h3">{props.header}</Text>
    </Box>
  );
};
