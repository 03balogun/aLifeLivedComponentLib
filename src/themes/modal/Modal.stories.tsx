import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { Modal } from './Modal';
import { ModalHeader } from './ModalHeader';
import { ModalFooter } from './ModalFooter';
import { Button } from '../../components/button/Button';
import { ModalContent } from './ModalContent';

storiesOf('Modal', module)
  .addDecorator(Story => Story())
  .add('Modal with header only', () => (
    <Modal onPress={action('Modal action fired')}>
      <ModalHeader header="This is testing the modal" />
      <ModalContent header="Test content" />
      <ModalFooter>
        <Button
          title="Test button one"
          onPress={action('Fire on press for button one')}
          width="50%"
          backgroundColor="primary"
          color="primary"
          marginRight="s"
        />
        <Button
          title="Test button two"
          onPress={action('Fire on press for button one')}
          width="50%"
          marginLeft="s"
        />
      </ModalFooter>
    </Modal>
  ));
