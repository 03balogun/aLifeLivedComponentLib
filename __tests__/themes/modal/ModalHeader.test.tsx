import React from 'react';
import { render, fireEvent } from '../../../jest/renderUtil';

// Components
import { ModalHeader } from '../../../src/themes/modal/ModalHeader';

const mockOnHeaderPress = jest.fn();

const setup = () => {
  const utils = render(<ModalHeader onHeaderPress={mockOnHeaderPress} />);

  return {
    ...utils,
  };
};

describe('themes >> ModalHeader', () => {
  describe('Render modal header', () => {
    test('Should render modal header correctly', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should fire onHeaderPress()', () => {
      const { getByText } = setup();
      fireEvent.press(getByText('X'));
      expect(mockOnHeaderPress).toBeCalledTimes(1);
    });
  });
});
