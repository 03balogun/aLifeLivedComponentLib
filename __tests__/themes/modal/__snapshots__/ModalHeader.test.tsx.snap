// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> ModalHeader Render modal header Should render modal header correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "flex-start",
        "display": "flex",
        "margin": 8,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#f4f5f5",
            "display": "flex",
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
        ]
      }
    >
      X
    </Text>
  </View>
</View>
`;
